src/Validator/Spec/Tag/LinkRelStylesheetForFonts.php

```diff
@@ -33,33 +49,22 @@ final class LinkRelStylesheetForFonts extends Tag
         SpecRule::SPEC_NAME => 'link rel=stylesheet for fonts',
         SpecRule::MANDATORY_PARENT => Element::HEAD,
         SpecRule::ATTRS => [
-            [
-                SpecRule::NAME => Attribute::ASYNC,
-            ],
-            [
-                SpecRule::NAME => Attribute::CROSSORIGIN,
-            ],
-            [
-                SpecRule::NAME => Attribute::HREF,
+            Attribute::ASYNC => [],
+            Attribute::CROSSORIGIN => [],
+            Attribute::HREF => [
                 SpecRule::MANDATORY => true,
                 SpecRule::VALUE_REGEX => 'https://cdn\.materialdesignicons\.com/([0-9]+\.?)+/css/materialdesignicons\.min\.css|https://cloud\.typography\.com/[0-9]*/[0-9]*/css/fonts\.css|https://fast\.fonts\.net/.*|https://fonts\.googleapis\.com/css2?\?.*|https://fonts\.googleapis\.com/icon\?.*|https://fonts\.googleapis\.com/earlyaccess/.*\.css|https://maxcdn\.bootstrapcdn\.com/font-awesome/([0-9]+\.?)+/css/font-awesome\.min\.css(\?.*)?|https://(use|pro|kit)\.fontawesome\.com/releases/v([0-9]+\.?)+/css/[0-9a-zA-Z-]+\.css|https://(use|pro|kit)\.fontawesome\.com/[0-9a-zA-Z-]+\.css|https://use\.typekit\.net/[\w\p{L}\p{N}_]+\.css',
```

Comment:
Since this regex will be sent to `preg_match()`, we should escape the '/' delimiter. See https://github.com/ampproject/amp-wp/pull/6303/commits/afd117289ee863011cb1a7dc4e096b49bd4a5f15

Reply:
I wouldn't escape this against a random delimiter.
I would suggest using the `chr(0)` delimiter in the plugin if you want to avoid `preg_quote()`.

Reply:
I'm merging this now to get rid of another huge PR, but we can discuss this particular issue separately.

Reply:
Interesting. I wasn't aware you could use a NUL as a delimiter.

Reply:
Seems you can't: 

![image](https://user-images.githubusercontent.com/134745/121562181-0fc5cf80-c9ce-11eb-9eaa-a7e3c2181b73.png)

Reply:
I've opened this as #238.

Reply:
Oh, sorry, I've remembered incorrectly, it was `chr(1)` that we have been using in WP-CLI, not `chr(0)`. The `chr(1)` character is the "start of header" (`SOH`) character, intended to control the flow of streams, and shouldn't be in use or cause conflicts in HTML.

This delimiter has been used as the default delimiter in the WP-CLI `search-replace` command when you provide a regex via the CLI, and as far as I know, it has never caused any issues.

=> https://3v4l.org/piT1j
![Online PHP editor _ output for piT1j - Google Chrome 2021-06-14 at 7 51 44 AM](https://user-images.githubusercontent.com/83631/121845203-0180f780-ccdd-11eb-83f8-c8d878b6006d.jpeg)

Reply:
Excellent. Sounds good to me.